build/gulp/version.js

```diff
@@ -0,0 +1,50 @@
+const getDefaultFlavor = (label, minor, revision) => {
+    return label
+        ? (minor <= 2 ? 'beta' : '')
+        : revision
+            ? (minor <= 1 ? 'alpha' : 'build')
+            : 'dev';
+};
+
+const getPackageVersion = (version, flavor, revision) => {
+    if(revision) {
+        version = version.replace(/\d+$/, m => 1 + Number(m));
+    }
+
+    return [version, flavor, revision]
+        .filter(v => v)
+        .join('-');
+};
+
+const getProductVersion = (version, revision) => {
+    return revision
+        ? version + ' (build ' + revision + ')'
+        : version;
+};
+
+const getRevision = (revision) => {
+    if(!revision) return '';
+
+    const date = new Date();
+    const padStart = (number) => ('0' + number).slice(-2);
+
+    const hours = padStart(date.getHours());
```

Comment:
How do we plan to sync this tail with NuGet packages?

Reply:
We'll get the version from  the `package.json` of npm package